etcdctl/ctlv3/command/check.go

```diff
@@ -157,13 +158,34 @@ func newCheckPerfCommand(cmd *cobra.Command, args []string) {
 	}
 
 	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(cfg.duration)*time.Second)
-	resp, err := clients[0].Get(ctx, checkPerfPrefix, v3.WithPrefix(), v3.WithLimit(1))
-	cancel()
+	defer cancel()
+
+	signalChan := make(chan os.Signal, 1)
+	defer signal.Stop(signalChan)
+	signal.Notify(signalChan, os.Interrupt)
+
+	go func() {
```

Comment:
Maybe move to a separate context: 

Ideally the integration could be based on invocation in following style: 

```
ctx := interruptableContext(ctx, func() { attemptCleanup(clients[0]) } )
```

Reply:
Thanks for the great suggestion, I think I implemented what you had in mind, please have a look, thank you!

Reply:
Thank you for fixing.